pkgs/applications/misc/far2l/default.nix

```diff
@@ -1,29 +1,34 @@
 { stdenv, fetchFromGitHub, makeWrapper, cmake, pkgconfig, wxGTK30, glib, pcre, m4, bash,
-  xdg_utils, gvfs, zip, unzip, gzip, bzip2, gnutar, p7zip, xz, imagemagick }:
+  xdg_utils, gvfs, zip, unzip, gzip, bzip2, gnutar, p7zip, xz, imagemagick, darwin }:
 
+with stdenv.lib;
 stdenv.mkDerivation rec {
-  rev = "de5554dbc0ec69329b75777d4a3b2f01851fc5ed";
-  build = "unstable-2017-07-13.git${builtins.substring 0 7 rev}";
+  rev = "1ecd3a37c7b866a4599c547ea332541de2a2af26";
+  build = "unstable-2017-09-30.git${builtins.substring 0 7 rev}";
   name = "far2l-2.1.${build}";
 
   src = fetchFromGitHub {
     owner = "elfmz";
     repo = "far2l";
     rev = rev;
-    sha256 = "07l8w9p6zxm9qgh9wlci584lgv8gd4aw742jaqh9acgkxy9caih8";
+    sha256 = "0mavg9z1n81b1hbkj320m36r8lpw28j07rl1d2hpg69y768yyq05";
   };
 
   nativeBuildInputs = [ cmake pkgconfig m4 makeWrapper imagemagick ];
 
-  buildInputs = [ wxGTK30 glib pcre ];
+  buildInputs = [ wxGTK30 glib pcre ]
+    ++ optional stdenv.isDarwin darwin.apple_sdk.frameworks.Cocoa;
 
   patches = [ ./add-nix-syntax-highlighting.patch ];
 
-  postPatch = ''
-    echo 'echo ${build}' > far2l/bootstrap/scripts/vbuild.sh
-
-    substituteInPlace far2l/bootstrap/open.sh              \
+  postPatch = optionalString stdenv.isLinux ''
+    substituteInPlace far2l/bootstrap/open.sh \
       --replace 'gvfs-trash'  '${gvfs}/bin/gvfs-trash'
+  '' + optionalString stdenv.isDarwin ''
+    substituteInPlace far2l/CMakeLists.txt \
+      --replace "-framework System" -lSystem
```

Comment:
Is this still necessary if you include `darwin.apple_sdk.frameworks.System`?

Reply:
No idea. I have absolutely no dev and almost no user experience on Darwin, just fixed "library not found" [after](https://github.com/NixOS/nixpkgs/blob/7ebcd39a0f147ffa544818c9fd5954bc9dcae2d8/pkgs/development/libraries/wxwidgets/3.0/default.nix#L69) [wxWidgets](https://github.com/NixOS/nixpkgs/blob/7ebcd39a0f147ffa544818c9fd5954bc9dcae2d8/pkgs/development/libraries/wxwidgets/3.0/mac.nix#L63) which ```far2l``` is using

Reply:
And... there is no ```System``` in ```darwin.apple_sdk.frameworks```: https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/darwin/apple-sdk/frameworks.nix

Reply:
oh, I thought I added that recently